src/asmcnc/apps/upgrade_app/screen_upgrade.py

```diff
@@ -247,49 +263,66 @@ def upgrade_and_proceed(self):
             log("Failed to create SC2 compatibility file!")
 
     def update_spindle_cooldown_settings(self):
-        # Write default SC2 settings, and set voltage to whatever is already selected
-        if not (self.m.write_spindle_cooldown_rpm_override_settings(False) and \
-            self.m.write_spindle_cooldown_settings(brand='YETI SC2', voltage=self.m.spindle_voltage, digital=True, time_seconds=10, rpm=self.m.yeti_cooldown_rpm_default)):
-
-            popup_info.PopupError(self.sm, self.l, self.l.get_str("There was a problem saving your settings."))
+        if not (
+            self.m.write_spindle_cooldown_rpm_override_settings(False)
+            and self.m.write_spindle_cooldown_settings(
+                brand="YETI SC2",
+                voltage=self.m.spindle_voltage,
+                digital=True,
+                time_seconds=10,
+                rpm=self.m.yeti_cooldown_rpm_default,
+            )
+        ):
+            popup_info.PopupError(
```

Comment:
Note that this popup will need to be come back to after Archie's new popup class has been merged

Reply:
Will this specific instance need testing, if this is just a generic error popup?

Reply:
Yes, it'll need to be updated with the new class to scale correctly :)